integration-test/1491-german-shields.py

```diff
@@ -0,0 +1,463 @@
+        import dsl
+
+        z, x, y = (16, 34437, 22569)
+
+        self.generate_fixtures(
+            dsl.is_in('DE', z, x, y),
+            # https://www.openstreetmap.org/way/13235013
+            dsl.way(13235013, dsl.tile_diagonal(z, x, y), {
+                'maxspeed': '50', 'placement': 'right_of:2',
+                'name': u'Roteb\xfchlstra\xdfe', 'surface': 'asphalt',
+                'source': 'openstreetmap.org', 'oneway': 'yes', 'lanes': '4',
+                'sidewalk': 'none', 'ref': 'B 27a', 'highway': 'primary',
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 13235013,
+                'shield_text': 'B27a', 'network': 'DE:BS',
+            })
+
+    def test_de_ring(self):
+        import dsl
+
+        z, x, y = (16, 34578, 21177)
+
+        self.generate_fixtures(
+            dsl.is_in('DE', z, x, y),
+            # https://www.openstreetmap.org/way/4644427
+            dsl.way(4644427, dsl.tile_diagonal(z, x, y), {
+                'maxspeed': '50', 'lanes': '2', 'name': 'Holstenplatz',
+                'source': 'openstreetmap.org', 'maxheight': 'none',
+                'cycleway:right': 'track', 'surface': 'asphalt',
+                'oneway': 'yes', 'sidewalk': 'right', 'ref': 'Ring 2',
+                'highway': 'secondary',
+            }),
+            dsl.relation(1, {
+                'name': 'Ring 2', 'ref': 'Ring 2', 'route': 'road',
+                'wikipedia': 'de:Ring 2 (Hamburg)',
+                'source': 'openstreetmap.org', 'wikidata': 'Q20731702',
+                'type': 'route',
+            }, ways=[4644427]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 4644427,
+                'shield_text': 'Ring 2', 'network': 'DE:Hamburg:Ring'})
+
+    def test_de_bs_ring_order(self):
+        import dsl
+
+        z, x, y = (16, 34583, 21151)
+
+        self.generate_fixtures(
+            dsl.is_in('DE', z, x, y),
+            # https://www.openstreetmap.org/way/4329503
+            dsl.way(4329503, dsl.tile_diagonal(z, x, y), {
+                'toll:N3': 'yes', 'maxspeed': '60', 'lanes': '2',
+                'name': 'Krohnstieg', 'surface': 'asphalt', 'lit': 'yes',
+                'source': 'openstreetmap.org', 'oneway': 'yes', 'foot': 'no',
+                'bicycle': 'no', 'ref': 'Ring 3;B 433', 'highway': 'trunk',
+            }),
+            dsl.relation(1, {
+                'source': 'openstreetmap.org', 'route': 'road',
+                'type': 'route', 'name': 'Ring 3',
+            }, ways=[4329503]),
+            dsl.relation(3, {
+                'name': 'B433', 'type': 'route', 'route': 'road',
+                'source': 'openstreetmap.org', 'ref': 'B 433',
+            }, ways=[4329503]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 4329503,
+                'shield_text': 'B433', 'network': 'DE:BS',
```

Comment:
The ref order makes the ring road seem more important than the general B road. Please confirm and if so switch the sort order?

Reply:
It doesn't seem to be consistent. For example, in the screenshot below it flips between `Ring 2; B5` and `B 5; Ring 2`.

![image](https://user-images.githubusercontent.com/271360/40741030-42cab3cc-6442-11e8-9173-534a8c7f3d42.png)

We can do it either way around - which would you prefer?